src/test/mod.rs

```diff
@@ -41,14 +40,14 @@ fn initialize(id: usize, root_path: Option<String>) -> Request<ls_server::Initia
 fn initialize_with_opts(
     id: usize,
     root_path: Option<String>,
-    initialization_options: Option<InitializationOptions>,
+    initialization_options: Option<serde_json::Value>,
```

Comment:
Not sure I understand this change - can't we continue to use `InitializationOptions` since it also has a `settings: Option<ChangeConfigSettings>` now? I would prefer not to additionally force-test (de)serialization here, since we have access to underlying structures in these "unit" tests

Reply:
If you look a little closer you'll see no addition deserialization has been added. The tests always deserialized because this method simply generates a `Request` that we send as a string to the server.

This is correct as clients send json strings to rls.

```rust
fn initialize_with_opts(
    id: usize,
    root_path: Option<String>,
    initialization_options: Option<InitializationOptions>,
) -> Request<ls_server::InitializeRequest> {
    let init_opts = initialization_options.map(|val| serde_json::to_value(val).unwrap());
    ...
}
```

However, what the test used to do is "force-test" a derived serialization implementation that was not used anywhere else. This implementation didn't handle `Config` well, so along with our poor serde failure logging made the new test fail for no apparent reason.

Deriving `Serialize` isn't actually any more convenient compared to `json!` usage. The latter is simpler, closer to actual client usage and less error prone while reducing prod code.